nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/test/java/org/nuxeo/elasticsearch/test/TestElasticSearchQuery.java

```diff
@@ -173,6 +176,45 @@ public void searchWithNestedQuery() throws Exception {
 
     }
 
+    @Test
+    public void testMoreLikeThisQuery() throws Exception {
+
+        //Create test data
+        startTransaction();
+        DocumentModel doc = session.createDocumentModel("/", "myFile", "File");
+        doc.setPropertyValue("dc:title", "very nice title");
+        doc = session.createDocument(doc);
+
+        for (int i = 1; i <= 5; i++) {
+            DocumentModel likeDoc = session.createDocumentModel("/", "myFile" + i, "File");
+            String title = i % 2 == 0 ? "nice title" : "nice";
+            likeDoc.setPropertyValue("dc:title", title + i);
+            session.createDocument(likeDoc);
+        }
+        TransactionHelper.commitOrRollbackTransaction();
+        waitForCompletion();
+        startTransaction();
+
+        //Query More Like This
+        MoreLikeThisQueryBuilder.Item item = new MoreLikeThisQueryBuilder.Item(null, null, doc.getId());
+        QueryBuilder elasticBuilder = QueryBuilders.moreLikeThisQuery(new String[] { "dc:title.fulltext" }, null,
+                new MoreLikeThisQueryBuilder.Item[] { item })
+                                                   .minTermFreq(1)
+                                                   .minWordLength(3)
+                                                   .maxQueryTerms(20)
+                                                   .minimumShouldMatch("45%");
+
+        NxQueryBuilder nxQuery = new NxQueryBuilder(session).esQuery(elasticBuilder)
+                                                            .fetchFromElasticsearch()
+                                                            .hitDocConsumer((searchHit, aDoc) -> {
+                                                                aDoc.putContextData(SCORE, searchHit.getScore());
```

Comment:
I had to reach this line to understand the purpose of the commit, the NXP don't give any clue about what it is.
So you enable EsFetcher to enrich document fetched from Elasticsearch with search hit info.
For sure it can be handy but this is limited to the none default EsFetch wich is experimental:
- it does not solve perf problem
- it lacks support of complex properties that are still loaded from the repository
May be this enricher helper could have been done on EsResult returned by queryAndAggregate where both DocumentModelList and elasticsearch SearchResponse are available.

Reply:
Thanks for the info.  Its good to learn that EsFetcher is experimental.  I didn't find anywhere in the code/javadoc to suggest this.  If its not intended for general use then we should state that somewhere, otherwise developers have no idea.